src/quo2/components/selectors/react_selector/style.cljs

```diff
@@ -0,0 +1,30 @@
+(ns quo2.components.selectors.react-selector.style
+  (:require [quo2.foundations.colors :as colors]))
+
+(defn- pinned-colors
+  [theme]
+  {:background-color (colors/theme-colors colors/neutral-80-opa-5
+                                          colors/white-opa-5
+                                          theme)
+   :border-color     (colors/theme-colors colors/neutral-80-opa-5
+                                          colors/white-opa-5
+                                          theme)})
+
+(defn reaction
+  [neutral? pinned? theme]
+  (cond-> {:flex-direction     :row
+           :justify-content    :center
+           :align-items        :center
+           :padding-horizontal 8
+           :border-radius      8
+           :height             24
+           :border-width       1
+           :border-color       (colors/theme-colors colors/neutral-20
+                                                    colors/neutral-80
+                                                    theme)
+           :background-color   (colors/theme-colors colors/neutral-10
+                                                    colors/neutral-80-opa-40)}
+    pinned?  (merge (pinned-colors theme))
```

Comment:
is merge needed here? 🤔

Reply:
thanks! was moving things around and didn't notice it.

Reply:
oh actually nevermind. The merge is needed because it's a conditional thread, so if `pinned?` is true, we merge `pinned-colors` to overwrite `:background-color` and `:border-color` on the original map. Unless I'm missing smth 🤔 ?

Reply:
I think I've seen a pattern where you don't need this. @ulisesmac maybe you know a nice way to write this :)

Reply:
Ok, I think it's a little hard, but there's room for some improvements.

1. `pinned-colors` is a private function with only one use, and it's inside `reaction`. 
2. last `merge` in `neutral?` can be avoided
3. the `pinned?` merge can also be avoided

So we can remove 
```clojure
(defn- pinned-colors ...)
```

and I found two alternatives:
1. preserve the `cond->`, avoid merges, no use of external function:
  ```clojure
  (defn reaction
  [neutral? pinned? theme]
  (let [pinned-color (colors/theme-colors colors/neutral-80-opa-5 colors/white-opa-5 theme)]
    (cond-> {:flex-direction     :row
             :justify-content    :center
             :align-items        :center
             :padding-horizontal 8
             :border-radius      8
             :height             24
             :border-width       1
             :border-color       (colors/theme-colors colors/neutral-20 colors/neutral-80 theme)
             :background-color   (colors/theme-colors colors/neutral-10 colors/neutral-80-opa-40)}
      pinned?  (assoc :background-color pinned-color :border-color pinned-color)
      neutral? (assoc :background-color :transparent))))
  ```
  
  2. no `cond->`, also no merges and no external function.
  ```clojure
  (let [pinned-color     (colors/theme-colors colors/neutral-80-opa-5 colors/white-opa-5 theme)
        border-color     (if pinned?
                           pinned-color
                           (colors/theme-colors colors/neutral-20 colors/neutral-80 theme))
        background-color (cond
                           pinned?  pinned-color
                           neutral? :transparent
                           :else    (colors/theme-colors colors/neutral-10 
                                                         colors/neutral-80-opa-40))]
    {:flex-direction     :row
     :justify-content    :center
     :align-items        :center
     :padding-horizontal 8
     :border-radius      8
     :height             24
     :border-width       1
     :border-color       border-color
     :background-color   background-color})
  ```
  
  Pick the one you like the most :)
  
  ### Note
  I noticed that the color passed to `:background-color` is not receiving `theme`, just wondering if it is expected or not:
  
```clojure
:background-color   (colors/theme-colors
                     colors/neutral-10
                     colors/neutral-80-opa-40
                     #_theme-not-passed-here)
  ```

Reply:
@ulisesmac I was trying to simplify it earlier, and ended up with `2` as well. I guess I'll go with that.

And thanks for catching the missing `theme`!